pkgs/tools/misc/glasgow/default.nix

```diff
@@ -46,24 +48,24 @@ python3.pkgs.buildPythonApplication rec {
     make -C firmware LIBFX2=${python3.pkgs.fx2}/share/libfx2
     cp firmware/glasgow.ihex software/glasgow
     cd software
-    export SETUPTOOLS_SCM_PRETEND_VERSION="${realVersion}"
+    export PDM_BUILD_SCM_VERSION="${realVersion}"
   '';
 
   # installCheck tries to build_ext again
   doInstallCheck = false;
 
   postInstall = ''
     mkdir -p $out/etc/udev/rules.d
-    cp $src/config/99-glasgow.rules $out/etc/udev/rules.d
+    cp $src/config/70-cypress.rules $out/etc/udev/rules.d
+    cp $src/config/70-glasgow.rules $out/etc/udev/rules.d
   '';
 
-  checkPhase = ''
+  preCheck = ''
     # tests attempt to cache bitstreams
     # for linux:
     export XDG_CACHE_HOME=$TMPDIR
     # for darwin:
     export HOME=$TMPDIR
```

Comment:
The PYTHONWARNINGS environment variable is able to silence the deprecation warnings ;)